src/can.rs

```diff
@@ -0,0 +1,190 @@
+//! Controller Area Network
+
+use nb;
+
+/// A type that can either be `BaseId` or `ExtendedId`
+#[cfg(feature = "unproven")]
+pub trait Id {
```

Comment:
I don't really get the value of not specifying concrete types (u16 and u32) for base and extended id.

How about adding a 

```rust
fn id(&self) -> u32;
```

possibly even with a blanket implementation based on the `base_id` and `extended_id` fns, but I think right now there is no way to guarantee that not both are not `None` so this might need to return `Result<u32, Error>` which is kind of ugly.